### ISSUE TITLE
Testing on Rawhide is too slow and times out

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#17866]
## TL/DR

Kernel regression, see https://bugzilla.redhat.com/show_bug.cgi?id=2139658

### Explain what happens

Rawhide runs regularly time out now, e.g. in PR #17864. I did some analysis:

## [successful F37 test](https://artifacts.dev.testing-farm.io/8673e4ce-739f-4870-bc30-0757ce6effea/)

- pipeline start:          18:14:15
- provision, pre-artifact: 18:17:32 + 3:17
- artifact install:        18:19:33 + 2:01
- post-artifact install:   18:19:34 + 0:01
- tmt starts:              18:20:00 + 0:26
- tmt ends:                18:52:05 + 22:05
- teardown, log upload:    18:53:27 + 1:22

package download is negligible (~ 1 s); package install is not measured, but 52
packages (with hopefully disabled fsync), shouldn't take that long (locally it's 37s for 176 packages, and no fsync mitigation)

    # TESTS PASSED [1654s on ip-172-31-30-193, 0 parallel tests, 44 serial tests: ]

→ this is 27:33 minutes, which is curious -- that's longer than tmt itself ran !?

First two tests:
```
# 1 TEST PASSED [18s on ip-172-31-30-193]
ok 36 test/verify/check-packagekit TestUpdates.testUpdateError [ND@-1]
# 1 TEST PASSED [20s on ip-172-31-30-193]
ok 34 test/verify/check-packagekit TestUpdates.testSecurityOnly [ND@-1]
```

## [failing rawhide test](https://artifacts.dev.testing-farm.io/a2dc57b6-7dd8-403b-b135-9b5947a75aaf/)

- pipeline start:          18:12:33
- provision, pre-artifact: 18:19:22 + 6:49
  (includes an [extensive upgrade/reboot](https://artifacts.dev.testing-farm.io/a2dc57b6-7dd8-403b-b135-9b5947a75aaf/guest-setup-afe07f94-6ec5-4e36-88a9-f54b6aa963b8/guest-setup-output-pre-artifact-installation.txt) due to old base image)
- artifact install:        18:21:59 + 2:27
- post-artifact install:   18:22:06 + 0:07
- tmt starts:              18:22:41 + 0:35
- tmt ends (timeout):      19:30:08 + 67:27
- teardown, log upload:    19:31:30 + 1:22

again, package download is negligible (3s), 53 test dependency installs
no run-tests summary as it got killed due to timeout

```
# 1 TEST PASSED [47s on ip-172-31-23-181]
ok 36 test/verify/check-packagekit TestUpdates.testUpdateError [ND@-1]
# 1 TEST PASSED [50s on ip-172-31-23-181]
ok 34 test/verify/check-packagekit TestUpdates.testSecurityOnly [ND@-1]
```
i.e. significantly slow (2.5x)

## local F37 run
In clean cockpit main checkout:

    tmt run --until report provision --how virtual --image fedora-37 prepare --how shell --script 'dnf install -y cockpit cockpit-storaged cockpit-tests'

```
14:42:16                 out: # 1 TEST PASSED [24s on testcloud]
14:42:16                 out: ok 36 test/verify/check-packagekit TestUpdates.testUpdateError [ND@-1]
14:42:41                 out: # 1 TEST PASSED [23s on testcloud]
14:42:41                 out: ok 34 test/verify/check-packagekit TestUpdates.testSecurityOnly [ND@-1]
```

That's just a tad slower than on TF, i.e. comparable.

## local rawhide run
In clean cockpit main checkout:

    tmt run --until report provision --how virtual --image fedora-rawhide prepare --how shell --script 'dnf install -y cockpit cockpit-storaged cockpit-tests'

Also similarly slow as on TF (slightly slower). With distro firefox (106):
```
14:16:49                 out: # 1 TEST PASSED [55s on testcloud]
14:16:49                 out: ok 36 test/verify/check-packagekit TestUpdates.testUpdateError [ND@-1]
14:17:49                 out: # 1 TEST PASSED [58s on testcloud]
14:17:49                 out: ok 34 test/verify/check-packagekit TestUpdates.testSecurityOnly [ND@-1]
```

With firefox nightly (108.0a1):
```
14:01:24                 out: # 1 TEST PASSED [45s on testcloud]
14:01:24                 out: ok 36 test/verify/check-packagekit TestUpdates.testUpdateError [ND@-1]
14:02:30                 out: # 1 TEST PASSED [64s on testcloud]
14:02:30                 out: ok 34 test/verify/check-packagekit TestUpdates.testSecurityOnly [ND@-1]
```

## Mitigation proposal

Roughly, setup costs are ~ 6 minutes (F37) / 10 minutes (rawhide), and actual tests are super slow on rawhide (> factor 5), so splitting them would make sense and is at least worth a try.

TMT runs each plan in a separate VM, up to 5 in parallel. So we could try to split the test into two or three batches.

## Root cause

The rawhide kernel [is known to be slow](https://lists.fedoraproject.org/archives/list/devel%40lists.fedoraproject.org/thread/HRJT4UF35MRJ7C4ZXVHBFXB4BTW64YSO/) due to having various debug options enabled. But that 5x slowdown is new. It's not specific to firefox, chromium, or cockpit, it affects literally everything on the system:

 - test with nightly upstream firefox instead of distro package: [F37](https://artifacts.dev.testing-farm.io/02a43a5c-3cc9-4ee4-be46-f0ed0748254a/) is similarly fast (26 min tmt time), [rawhide](https://artifacts.dev.testing-farm.io/8448ef1f-fab7-4ea5-b1f3-4442d652522b/) also times out; same locally; thus, no change.

 - test with chromium: faster locally (29s/30s for the first two tests); 25 min tmt runtime on [F37](https://artifacts.dev.testing-farm.io/99bc91f7-98f2-44f7-8c1e-7153ac75f1f9/), 54 min on [rawhide](https://artifacts.dev.testing-farm.io/2b528788-d7fa-471a-a917-9dd6a04debe1/); thus, better, but not much 
 
 - test cockpit CI style, i.e. browser on host, and fedora-rawhide image: can't run fedora-rawhide on CI, it [keeps timing out on shutdown](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17867-20221102-193756-fc10da6e-fedora-rawhide/log.html); same result locally, with
 
         TEST_OS=fedora-37 test/common/run-tests --test-dir test/verify TestUpdates.{testUpdateError,testSecurityOnly}

   vs. `TEST_OS=fedora-rawhide`. The tests above take 29s/20s on f37, and 68s/24s on f-rawhide.
 
 - even booting is significantly slower. F37:
   ```
   Startup finished in 811ms (kernel) + 1.037s (initrd) + 5.412s (userspace) = 7.262s
   multi-user.target reached after 4.969s in userspace.
   ```
   rawhide:
   ```
   Startup finished in 3.136s (kernel) + 3.082s (initrd) + 13.449s (userspace) = 19.669s 
   multi-user.target reached after 13.239s in userspace.
   ```

- Installing F37's kernel-core-5.19.16-301.fc37.x86_64.rpm and kernel-tools{,-libs} on rawhide entirely fixes this. The two tests above are fast, and boot speed is comparable to F37. So it's not rawhide userspace, it's kernel-core-6.1.0-0.rc1.20221019gitaae703b02f92.17.fc38.x86_64 .

- [PR history](https://github.com/cockpit-project/cockpit/pulls?page=2&q=is%3Apr+is%3Aclosed):
   * #17799 [ran on Oct 10](https://artifacts.dev.testing-farm.io/84b50512-347e-4475-9cc6-72e455ea36ce/work-allpeuDKR/log.txt.gz), kernel 6.1.0-0.rc0.20221007git4c86114194e6.5.fc38.x86_64, timed out
   * #17798 also [ran on Oct 10](https://artifacts.dev.testing-farm.io/e79d2743-0c53-4d29-bccd-010c129a6725/work-allQ8HHhU/log.txt.gz), same kernel as above, just barely finished (62 mins)
   * #17783 [ran on Sep 29](https://artifacts.dev.testing-farm.io/06ffcd55-0eea-411d-9045-71ca53f9aca6/work-allWyMHDE/log.txt.gz), kernel 6.0.0-0.rc7.20220928git49c13ed0316d.50.fc38.x86_64, just barely finished (57 minutes)
   * #17741 [ran on Sep 23](https://artifacts.dev.testing-farm.io/327c599a-9176-4bff-8d8d-eea02d8f1a62/work-allw4Udzv/log.txt.gz), kernel 6.0.0-0.rc6.41.fc38.x86_64, tmt took 30 mins
   * #17735 [ran on Sep 14](https://artifacts.dev.testing-farm.io/ea0723d2-4bf3-4ac5-b231-4c69b232d233//work-allxz3iXF/log.txt.gz), 6.0.0-0.rc5.37.fc38.x86_64 tmt took 30 mins
   * #17717 [ran on Sep 7](https://artifacts.dev.testing-farm.io/773e1fe1-3445-42fb-9caf-d01f65e665ae//work-allqqvr_Y/log.txt.gz), kernel 6.0.0-0.rc4.31.fc38.x86_64 tmt took just 27 mins

So I think we can blame this roughly on 6.0.0-0.rc7.

**Comment by martinpitt** [comment]
@marusak: FYI what we talked about this morning.

**Comment by martinpitt** [comment]
The description now contains all the data that I am able to squeeze out. I reported this at https://bugzilla.redhat.com/show_bug.cgi?id=2139658 as it's a genuine kernel bug.
